### PR TITLE
[FEATURE] Add data_store::validate

### PR DESCRIPTION
**Pull request by eseiler**
Part 1 of #227

### CODE REVIEW

**Comment by seqan-actions** [comment]
Documentation preview available at https://docs.seqan.de/preview/seqan/hibf/230

File: src/layout/data_store.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    if (hibf_layout == nullptr)
+        throw std::invalid_argument{"[HIBF ERROR] data_store::hibf_layout must not be nullptr."};
+
+    if (kmer_counts == nullptr)
+        throw std::invalid_argument{"[HIBF ERROR] data_store::kmer_counts must not be nullptr."};
+
+    if (sketches != nullptr && kmer_counts->size() != sketches->size())
+        throw std::invalid_argument{
+            "[HIBF ERROR] data_store::kmer_counts and data_store::sketches must have the same size."};
+
```
Comment (smehringer) [inline src/layout/data_store.cpp:21]:
```suggestion

   if (positions.size() > kmer_counts->size())
        throw std::invalid_argument{"[HIBF ERROR] data_store::positions must be smaller or euqal in size to kmer_counts."};

```

File: src/layout/hierarchical_binning.cpp [hunk 2]
```diff
@@ -40,7 +39,8 @@ size_t hierarchical_binning::execute()
 
         if (!config.disable_estimate_union && !config.disable_rearrangement)
         {
-            assert(data->sketches != nullptr);
+            if (data->sketches == nullptr)
+                throw std::invalid_argument{"[HIBF ERROR] data_store::sketches must not be nullptr."};
```
Comment (smehringer) [inline src/layout/hierarchical_binning.cpp:43]:
```suggestion
                throw std::invalid_argument{"[HIBF ERROR] data_store::sketches must not be nullptr is union estimation or rearrangement is enabled."};
```

**Review by smehringer — CHANGES_REQUESTED** [review]
tiny add ons

**Review by smehringer — APPROVED** [review]

**Pull request merged**